File: niro/src/main/scala/nl/uva/se/sc/niro/model/qls/style/CheckBox.scala
package nl.uva.se.sc.niro.model.qls.style

import nl.uva.se.sc.niro.model.ql.{ AnswerType, BooleanType }

case class CheckBox() extends WidgetType {
  override def isCompatibleWith(answerType: AnswerType): Boolean = answerType == BooleanType
}

File: qhwcz/QLS.Core/Validation/WidgetTypes/Checkbox.cs
using QL.Api.Entities;
using QLS.Api.Entities;

namespace QLS.Core.Validation.WidgetTypes
{
    public class Checkbox : IWidget
    {
        public WidgetType Type => WidgetType.Checkbox;

        public bool IsCompatibleWith(QLType questionType)
        {
            return questionType == QLType.Boolean;
        }

        public override string ToString()
        {
            return WidgetType.Checkbox.ToString();
        }
    }
}
